lib/saw/warehouse_test.go

```diff
@@ -274,6 +274,52 @@ func TestSAW_GetServiceAccounts(t *testing.T) {
 	}
 }
 
+func TestSAW_GetAccountKeyWithRoles_BQ(t *testing.T) {
+	ctx := context.Background()
+
+	fix, cleanup := newFix(t)
+	defer cleanup()
+
+	store := fakestore.New()
+	saw := New(store, fix.iam, fix.creds, fix.crm, fix.gcs, fix.bqds, nil)
+	params := &clouds.ResourceTokenCreationParams{
+		AccountProject: "fake-account-project",
+		Items:          []map[string]string{
+			{   "project": "fake-project-id",
+				"dataset": "fake-dataset",
+				"job-project": "fake-job-project",
```

Comment:
So I think we're missing an assertion in this test because I don't see `"fake-job-project"` used anywhere below. The most important assertion in this test will be asserting that we've added `roles/bigquery.user` to `"fake-job-project"` and not `"fake-project-id"`.